test/jvm/src/main/scala/zio/test/environment/TestEnvironment.scala

```diff
@@ -29,30 +30,33 @@ case class TestEnvironment(
   random: TestRandom.Test,
   scheduler: TestClock.Test,
   sized: Sized.Service[Any],
-  system: TestSystem.Test
+  system: TestSystem.Test,
+  testAnnotations: TestAnnotations.Service[Any]
 ) extends Blocking
     with Live[ZEnv]
+    with Scheduler
+    with Sized
+    with TestAnnotations
     with TestClock
     with TestConsole
     with TestRandom
     with TestSystem
-    with Scheduler
-    with Sized
 
 object TestEnvironment {
 
   val Value: Managed[Nothing, TestEnvironment] =
     Managed.fromEffect {
       for {
-        clock    <- TestClock.makeTest(TestClock.DefaultData)
-        console  <- TestConsole.makeTest(TestConsole.DefaultData)
-        live     <- Live.makeService(new DefaultRuntime {}.Environment)
-        random   <- TestRandom.makeTest(TestRandom.DefaultData)
-        time     <- live.provide(zio.clock.nanoTime)
-        _        <- random.setSeed(time)
-        size     <- Sized.makeService(100)
-        system   <- TestSystem.makeTest(TestSystem.DefaultData)
-        blocking = Blocking.Live.blocking
-      } yield new TestEnvironment(blocking, clock, console, live, random, clock, size, system)
+        clock           <- TestClock.makeTest(TestClock.DefaultData)
+        console         <- TestConsole.makeTest(TestConsole.DefaultData)
+        live            <- Live.makeService(new DefaultRuntime {}.Environment)
+        random          <- TestRandom.makeTest(TestRandom.DefaultData)
+        size            <- Sized.makeService(100)
+        testAnnotations <- TestAnnotations.makeService
```

Comment:
Don't we need (in the most general possible case), one `TestAnnotationMap` per test? Of course, we can compose all `TestAnnotation` inside a suite into one, for the suite itself, and so forth; but it seems to me to have the most flexibility, we will need one per test.

(This, btw, explains my confusion about the interface for `TestAnnotations`, specifically the utility about the `make*` methods.)

Reply:
Yes, I think we want one `TestAnnotationMap` per test. Right now I am doing this because the `TestExecutor` provides a separate copy of the `TestEnvironment` to each test. Does that make sense or would you approach it differently?

Reply:
Ah, I didn't see that. Where is that code?

Reply:
The `TestExecutor` uses `Spec#provideManaged` [here](https://github.com/zio/zio/blob/ef2375e434dce8d2e8fa6c9955ba03f77f822454/test/shared/src/main/scala/zio/test/TestExecutor.scala#L27) and `provideManaged` (as opposed to `provideManagedShared`) provides a separate copy of the resources to each test [here](https://github.com/zio/zio/blob/ef2375e434dce8d2e8fa6c9955ba03f77f822454/test/shared/src/main/scala/zio/test/Spec.scala#L186).

Reply:
So the only problem I see is that if the user chooses to share an environment across tests (e.g. to share database connection costs), then suddenly the annotations will not be per test, but rather, "global" per the sharing. 

It would be nice if we could guarantee the same annotations regardless of environmental sharing; and also, if the annotations for suites were computed through semigroup composition of the annotations of their contents (it's a stronger guarantee).

What do you think?

Reply:
I tried that initially and ran into some issues. For a lot of annotations we don't actually want to combine them using the semigroup for the test annotation. For example, for the `Timing` annotation it isn't that interesting to just add up the duration for all the tests, we want to combine them using more of a "map merge" semigroup with mappings from labels to durations. The more serious issue is that we want to write information to test annotations through test aspects but test aspects don't have access to test labels. So if we have each test write its duration to a shared annotation map we don't have any keys to associate with the durations in a sensible way to then aggregate.

Maybe another way we could address is by taking the `TestAnnotations` service out of the test environment and having the `TestExecutor` provide it separately to all the tests so it would be impossible or at least very hard for a user to provide a shared annotation map.

I do think the semigroup solution is the most natural one so would love to do that if possible but not sure how to overcome the issues there.

Reply:
I'll write up some ideas here soon!

Reply:
If we use a fiber ref, then during the evaluation of tests, the executor can create a fresh map, and replace the contents of the fiber ref with that fresh map, and then after the test, capture the map, and in the end, generate one map per test, and one map per suite. The suite maps would be generated by using the annotation append functionality, which merges collisions, which means that suite maps would always be an aggregation of their children.

Then the renderer can look at the map and decide what to render, per node, based on the desired rendering test annotations (timing, failure count, success count, etc.).